src/modules/cli/lib/install-macos.ts

```diff
@@ -27,52 +32,84 @@ export async function installCLIOnMacOSWithConfirmation() {
 		Sentry.captureException( error );
 		console.error( 'Failed to install CLI', error );
 
+		let message = __( 'Please ensure you grant Studio admin permissions when prompted.' );
+		if ( error instanceof Error && error.message === InstallError.FileAlreadyExists ) {
+			message = sprintf(
+				/* translators: 1: Installation path */
+				__(
+					'The installation path %1$s is already occupied by a file or directory. Please remove it and try again.'
+				),
+				cliSymlinkPath
+			);
+		}
+
```

Comment:
This fails to propagate legitimate errors occurred during execution of the sudo'd command. Try it yourself by exiting early from `install-studio-cli.sh` with `exit 1`, or even echoing an error and exiting with 1. The code assumes a failure will be due to permissions not being granted, but that can be verified by looking at `error.message`:

|Problem|`error.message`|
|-|-|
|Permission not granted|_User did not grant permission._|
|Non-zero exit with no echoed output|_Command failed: /bin/sh "/Applications/Studio.app/Contents/Resources/bin/install-studio-cli.sh"_|
|Non-zero exit with echoed output|_Command failed: /bin/sh "/Applications/Studio.app/Contents/Resources/bin/install-studio-cli.sh"`\n`Error: Foo_|

Reply:
This is true, but I think it's fair to assume that most errors generated here will be due to missing permissions. Moreover, the logic in `install-studio-cli.sh` is unlikely to generate errors (especially considering that we verify that `cliSymlinkPath` is not occupied before running the script). 

Still, I added some additional error message parsing here for completeness' sake, so we now also distinguish between known permission errors and unknown errors.